### ISSUE TITLE
Transfer learning and training without pre-trained model

### ISSUE THREAD

**Issue by ansl02** [ultralytics/ultralytics#4663]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello?

I am a beginner and i first encountered YOLO through yolov8.

I have a few questions and couldn't find them in the issue. 

1. Is the code below correct for training using only the yolov8n architecture without using a pre-training model? Is it training from the beginning?
Code: yolo detect train model=yolov8n.yaml data=./mydata.yaml epochs=10

2. Even if i input yolov8n.yaml and yolov8n.pt in the model, the phrase freezing layer 'model.22.dfl.conv.weight' appears. Why does it appear?

3. If i input model=yolov8n.pt, 319/355 items will be transferred. What does items mean here?

4. Transfer learning is performed by entering the code model=yolov8n.pt. Does this mean that even the front of the classifier is frozen and only the classifier is trained? Or do you just import the learned weights and biases, and then train the entire architecture?

5. What means AMP? Is it necessary?

6. Finally, i found that closing dataloader mosaic sentence. Is it conduct data augmentation mosaic method? If it correct, how stop data augmentation method?

Please help me...

If there is any incorrect information, please point it out.

I'm sorry for my poor English skills.

Thank you

### Additional

                   from  n    params  module                                       arguments                    
  0                  -1  1      1392  ultralytics.nn.modules.conv.Conv             [3, 48, 3, 2]                
  1                  -1  1     41664  ultralytics.nn.modules.conv.Conv             [48, 96, 3, 2]                
  2                  -1  2    111360  ultralytics.nn.modules.block.C2f             [96, 96, 2, True]            
  3                  -1  1    166272  ultralytics.nn.modules.conv.Conv             [96, 192, 3, 2]              
  4                  -1  4    813312  ultralytics.nn.modules.block.C2f             [192, 192, 4, True]          
  5                  -1  1    664320  ultralytics.nn.modules.conv.Conv             [192, 384, 3, 2]              
  6                  -1  4   3248640  ultralytics.nn.modules.block.C2f             [384, 384, 4, True]          
  7                  -1  1   1991808  ultralytics.nn.modules.conv.Conv             [384, 576, 3, 2]              
  8                  -1  2   3985920  ultralytics.nn.modules.block.C2f             [576, 576, 2, True]          
  9                  -1  1    831168  ultralytics.nn.modules.block.SPPF            [576, 576, 5]                
 10                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 11             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]                          
 12                  -1  2   1993728  ultralytics.nn.modules.block.C2f             [960, 384, 2]                
 13                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 14             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]                          
 15                  -1  2    517632  ultralytics.nn.modules.block.C2f             [576, 192, 2]                
 16                  -1  1    332160  ultralytics.nn.modules.conv.Conv             [192, 192, 3, 2]              
 17            [-1, 12]  1         0  ultralytics.nn.modules.conv.Concat           [1]                          
 18                  -1  2   1846272  ultralytics.nn.modules.block.C2f             [576, 384, 2]                
 19                  -1  1   1327872  ultralytics.nn.modules.conv.Conv             [384, 384, 3, 2]              
 20             [-1, 9]  1         0  ultralytics.nn.modules.conv.Concat           [1]                          
 21                  -1  2   4207104  ultralytics.nn.modules.block.C2f             [960, 576, 2]                
 22        [15, 18, 21]  1   3822016  ultralytics.nn.modules.head.Detect           [80, [192, 384, 576]]        
YOLOv8m summary: 295 layers, 25902640 parameters, 25902624 gradients

Ultralytics YOLOv8.0.166 🚀 Python-3.8.17 torch-2.0.1+cu117 CUDA:0 (NVIDIA GeForce RTX 4080, 16071MiB)
engine/trainer: task=detect, mode=train, model=yolov8m.yaml, data=./merged/nuimages/data.yaml, epochs=10, patience=50, batch=16, imgsz=640, save=True, save_period=-1, cache=False, device=None, workers=8, project=None, name=None, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=ture, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, show=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, vid_stride=1, stream_buffer=False, line_width=None, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, boxes=True, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0, cfg=None, tracker=botsort.yaml, save_dir=/home/ansl/ultralytics/runs/detect/train3
Overriding model.yaml nc=80 with nc=3

                   from  n    params  module                                       arguments                    
  0                  -1  1      1392  ultralytics.nn.modules.conv.Conv             [3, 48, 3, 2]                
  1                  -1  1     41664  ultralytics.nn.modules.conv.Conv             [48, 96, 3, 2]                
  2                  -1  2    111360  ultralytics.nn.modules.block.C2f             [96, 96, 2, True]            
  3                  -1  1    166272  ultralytics.nn.modules.conv.Conv             [96, 192, 3, 2]              
  4                  -1  4    813312  ultralytics.nn.modules.block.C2f             [192, 192, 4, True]          
  5                  -1  1    664320  ultralytics.nn.modules.conv.Conv             [192, 384, 3, 2]              
  6                  -1  4   3248640  ultralytics.nn.modules.block.C2f             [384, 384, 4, True]          
  7                  -1  1   1991808  ultralytics.nn.modules.conv.Conv             [384, 576, 3, 2]              
  8                  -1  2   3985920  ultralytics.nn.modules.block.C2f             [576, 576, 2, True]          
  9                  -1  1    831168  ultralytics.nn.modules.block.SPPF            [576, 576, 5]                
 10                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 11             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]                          
 12                  -1  2   1993728  ultralytics.nn.modules.block.C2f             [960, 384, 2]                
 13                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 14             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]                          
 15                  -1  2    517632  ultralytics.nn.modules.block.C2f             [576, 192, 2]                
 16                  -1  1    332160  ultralytics.nn.modules.conv.Conv             [192, 192, 3, 2]              
 17            [-1, 12]  1         0  ultralytics.nn.modules.conv.Concat           [1]                          
 18                  -1  2   1846272  ultralytics.nn.modules.block.C2f             [576, 384, 2]                
 19                  -1  1   1327872  ultralytics.nn.modules.conv.Conv             [384, 384, 3, 2]              
 20             [-1, 9]  1         0  ultralytics.nn.modules.conv.Concat           [1]                          
 21                  -1  2   4207104  ultralytics.nn.modules.block.C2f             [960, 576, 2]                
 22        [15, 18, 21]  1   3777433  ultralytics.nn.modules.head.Detect           [3, [192, 384, 576]]          
YOLOv8m summary: 295 layers, 25858057 parameters, 25858041 gradients

Freezing layer 'model.22.dfl.conv.weight'
AMP: running Automatic Mixed Precision (AMP) checks with YOLOv8n...
AMP: checks passed ✅
train: Scanning /home/ansl/ultralytics/merged/nuimages/train/labels.cache... 50 images, 0 backgrounds, 0 corrupt: 100%|██████████| 50/
val: Scanning /home/ansl/ultralytics/merged/nuimages/test/labels... 50 images, 0 backgrounds, 0 corrupt: 100%|██████████| 50/50 [00:00
val: New cache created: /home/ansl/ultralytics/merged/nuimages/test/labels.cache
Plotting labels to /home/ansl/ultralytics/runs/detect/train3/labels.jpg...
optimizer: AdamW(lr=0.001429, momentum=0.9) with parameter groups 77 weight(decay=0.0), 84 weight(decay=0.0005), 83 bias(decay=0.0)
Image sizes 640 train, 640 val
Using 8 dataloader workers
Logging results to /home/ansl/ultralytics/runs/detect/train3
Starting training for 10 epochs...
Closing dataloader mosaic

**Comment by geraldinedeeplearning** [comment]
Have you by any chance found out why layer `'model.22.dfl.conv.weight'` is frozen?

**Comment by glenn-jocher** [comment]
@geraldinedeeplearning the layer 'model.22.dfl.conv.weight' is likely frozen to maintain the weights that were pre-trained on a larger dataset such as COCO or ImageNet. Freezing layers is a common practice in transfer learning, where a model trained on one task is reused as the starting point for a model on a second task.

For YOLOv8, certain layers like the 'model.22.dfl.conv.weight' may be frozen during training to prevent these layers from updating. This is useful for preserving high-level feature detectors learned by these layers. These kinds of layers are usually early convolutional layers in the network that capture general features. Updating them can sometimes have a negative impact on your model's performance, which is why they're frozen.

You can modify which layers to freeze during training if required. Keep in mind that freezing and unfreezing layers is a strategic decision and should be handled with care to achieve the right balance between model performance and training efficiency.
